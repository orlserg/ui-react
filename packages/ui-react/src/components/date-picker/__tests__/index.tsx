import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { ru } from 'date-fns/locale';

import { DatePicker } from '..';

// TODO дописать тесты когда-нибудь

const i18nValues = {
    'start-date': 'Начало периода',
    'end-date': 'Конец периода',
    'apply': 'Применить',
    'whole-period': 'Весь период'
};

test('Default use', async () => {
    const { asFragment } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Range Picker', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            isDateRange={true}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Range Picker With Dates', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            isDateRange={true}
            defaultSelectedDate={[new Date('2023-06-25'), new Date('2023-06-27')]}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Single With Dates', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            defaultSelectedDate={[new Date('2023-06-25'), new Date('2023-06-27')]}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Default Min Max Dates', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            defaultMinDate={new Date('2023-06-25')}
            defaultMaxDate={new Date('2023-06-27')}
            defaultSelectedDate={[new Date('2023-06-25'), new Date('2023-06-27')]}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Default Min Max Dates', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            disableDatesInPast={true}
            defaultMinDate={new Date('2023-06-25')}
            defaultMaxDate={new Date('2023-06-27')}
            defaultSelectedDate={[new Date('2023-06-25'), new Date('2023-06-27')]}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Default Min Max Dates Period', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            disableDatesInPast={true}
            defaultMinDate={new Date('2023-06-25')}
            defaultMaxDate={new Date('2023-06-27')}
            defaultSelectedDate={[new Date('2023-06-25')]}
            disabledDates={[new Date('2023-06-25'), new Date('2023-06-25')]}
            disabled={true}
            maxPeriodDays={14}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Same period', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            defaultSelectedDate={[new Date('2023-06-25'), new Date('2023-06-25')]}
            readOnly={true}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Date Range Is Mobile', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            isMobile={true}
            isDateRange={true}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Datepicker Is Mobile', async () => {
    const { asFragment, getByTitle } = render(
        <DatePicker
            i18nConfig={{
                locale: ru,
                translation: i18nValues
            }}
            isMobile={true}
            qa={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByTitle('date-picker'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

