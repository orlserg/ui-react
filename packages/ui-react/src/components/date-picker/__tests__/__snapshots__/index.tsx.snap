// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Range Is Mobile 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <button
      class="date-picker__trigger date-picker__trigger_range"
      title="date-picker"
      type="button"
    >
      <p
        class="typography typography_size-body typography_color-basic typography_no-spacing"
      />
      <svg
        class="date-picker__field-icon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Date Range Is Mobile 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <button
      class="date-picker__trigger date-picker__trigger_range"
      title="date-picker"
      type="button"
    >
      <p
        class="typography typography_size-body typography_color-basic typography_no-spacing"
      />
      <svg
        class="date-picker__field-icon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Datepicker Is Mobile 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <button
      class="date-picker__trigger"
      title="date-picker"
      type="button"
    >
      <p
        class="typography typography_size-body typography_color-basic typography_no-spacing"
      />
      <svg
        class="date-picker__field-icon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Datepicker Is Mobile 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <button
      class="date-picker__trigger"
      title="date-picker"
      type="button"
    >
      <p
        class="typography typography_size-body typography_color-basic typography_no-spacing"
      />
      <svg
        class="date-picker__field-icon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates 3`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates 4`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates Period 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default Min Max Dates Period 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default use 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        />
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        />
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Range Picker 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger date-picker__trigger_range"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        />
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Range Picker 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger date-picker__trigger_range"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        />
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Range Picker With Dates 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger date-picker__trigger_range"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023 - 27.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Range Picker With Dates 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger date-picker__trigger_range"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023 - 27.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Same period 1`] = `
<DocumentFragment>
  <button
    class="date-picker__trigger"
    title="date-picker"
    type="button"
  >
    <p
      class="typography typography_size-body typography_color-basic typography_no-spacing"
    >
      25.06.2023
    </p>
    <svg
      class="date-picker__field-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Same period 2`] = `
<DocumentFragment>
  <button
    class="date-picker__trigger"
    title="date-picker"
    type="button"
  >
    <p
      class="typography typography_size-body typography_color-basic typography_no-spacing"
    >
      25.06.2023
    </p>
    <svg
      class="date-picker__field-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Single With Dates 1`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Single With Dates 2`] = `
<DocumentFragment>
  <div
    class="date-picker"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="date-picker__trigger"
        title="date-picker"
        type="button"
      >
        <p
          class="typography typography_size-body typography_color-basic typography_no-spacing"
        >
          25.06.2023
        </p>
        <svg
          class="date-picker__field-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 6H6V5H4v3h16V5h-2v1h-2V5H8v1Zm12 4H4v10h16V10Zm-4-7H8V2H6v1H4a2 2 0 0 0-2 2v15a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-2V2h-2v1ZM7 14v-2h2v2H7Zm4 0h2v-2h-2v2Zm4 0v-2h2v2h-2Zm-8 2v2h2v-2H7Zm6 2h-2v-2h2v2Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </span>
  </div>
</DocumentFragment>
`;
